src/main/scala/rocket/RVC.scala

```diff
@@ -60,15 +62,24 @@ class RVCDecoder(x: UInt, xLen: Int, useAddiForMv: Boolean = false) {
       if (xLen == 32) inst(Cat(lwImm, rs1p, 2.U(3.W), rs2p, 0x07.U(7.W)), rs2p, rs1p, rs2p)
       else ld
     }
-    def unimp = inst(Cat(lwImm >> 5, rs2p, rs1p, 2.U(3.W), lwImm(4,0), 0x3F.U(7.W)), rs2p, rs1p, rs2p)
+    def zcb_q0 = {
+      def lbu = Cat(lbImm, rs1p, 4.U(3.W), rs2p, 0x03.U(7.W))
+      def lh  = {
+        val func3 = Mux(x(6), 1.U(3.W), 5.U(3.W))
+        Cat(lhImm, rs1p, func3, rs2p, 0x03.U(7.W))
+      }
+      def sb  = Cat(rs2p, rs1p, 0.U(3.W), 0.U(3.W), lbImm(1,0), 0x23.U(7.W))
+      def sh  = Cat(rs2p, rs1p, 1.U(3.W), 0.U(3.W), lhImm(1,0), 0x23.U(7.W))
+      inst(Seq(lbu, lh, sb, sh)(x(11,10)), rs2p, rs1p, rs2p)
+    }
     def sd = inst(Cat(ldImm >> 5, rs2p, rs1p, 3.U(3.W), ldImm(4,0), 0x23.U(7.W)), rs2p, rs1p, rs2p)
     def sw = inst(Cat(lwImm >> 5, rs2p, rs1p, 2.U(3.W), lwImm(4,0), 0x23.U(7.W)), rs2p, rs1p, rs2p)
     def fsd = inst(Cat(ldImm >> 5, rs2p, rs1p, 3.U(3.W), ldImm(4,0), 0x27.U(7.W)), rs2p, rs1p, rs2p)
     def fsw = {
       if (xLen == 32) inst(Cat(lwImm >> 5, rs2p, rs1p, 2.U(3.W), lwImm(4,0), 0x27.U(7.W)), rs2p, rs1p, rs2p)
       else sd
     }
-    Seq(addi4spn, fld, lw, flw, unimp, fsd, sw, fsw)
+    Seq(addi4spn, fld, lw, flw, zcb_q0, fsd, sw, fsw)
```

Comment:
Zcb should be populated with conditions

For example, `c.zext.b` should be added only when `usingBitManip`, and `c.mul` should be populated only when `usingMulDiv`, `zext.w` should be added only when RV64. This could save a tiny bit of area.

Other insts like `c.lbu` and `c.not` will always be added as `I` or `E` are always present. This default behavior will add area overhead for end users as they may not use it. I think we provide an option like `usingCompressedBitManip` to control this behavior. Asking @jerryrzhao for advice.

Reply:
A mistake. Should @jerryz123.

Reply:
Parameters to the RVCDecode object should be added that disable the Zcb/Zcmp/Zcmt extensions if not needed.